.github/workflows/formatter.yml

```diff
@@ -0,0 +1,37 @@
+name: Format
+
+on:
+  pull_request:
+    paths: ['**.cs']
+  push:
+    paths: ['**.cs']
+
+permissions:
+  contents: write
+
+jobs:
+  format:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Setup .NET
+        uses: actions/setup-dotnet@v3
+        with:
+          dotnet-version: 7.0.x
+
+      - name: Format scripts
+        run: dotnet tool run dotnet-format --fix-whitespace
+
+      - name: Set Git config
+        run: |
+          git config user.name 'octocat'
+          git config user.email 'octocat@users.noreply.github.com'
+
+      - name: Commit changes
+        run: |
+          git add .
+          git commit -m "style: fix whitespaces" | true
+          git push
```

Comment:
This is very neat, but it didn't work for me on a PR (`fatal: You are not currently on a branch.`).
You need to fix it, or  use "auto-commit" action instead:
```
      - uses: stefanzweifel/git-auto-commit-action@v4
        with:
          file_pattern: "**.cs"
          commit_user_name: octocat
          commit_user_email: octocat@users.noreply.github.com
          commit_message: "style: fix whitespaces"
```

Reply:
I try to avoid using external actions because of the performance overhead, so I'll just fix it.